pkg/controllers/disruption/singlenodeconsolidation.go

```diff
@@ -111,3 +151,98 @@ func (s *SingleNodeConsolidation) Class() string {
 func (s *SingleNodeConsolidation) ConsolidationType() string {
 	return SingleNodeConsolidationType
 }
+
+func (s *SingleNodeConsolidation) groupCandidatesByNodePool(candidates []*Candidate) (map[string][]*Candidate, []string) {
+	nodePoolCandidates := make(map[string][]*Candidate)
+	nodePoolNames := []string{}
+
+	for _, candidate := range candidates {
+		nodePoolName := candidate.nodePool.Name
+		if _, exists := nodePoolCandidates[nodePoolName]; !exists {
+			nodePoolNames = append(nodePoolNames, nodePoolName)
+		}
+		nodePoolCandidates[nodePoolName] = append(nodePoolCandidates[nodePoolName], candidate)
+	}
+	return nodePoolCandidates, nodePoolNames
+}
+
+func (s *SingleNodeConsolidation) sortNodePoolsByTimeout(ctx context.Context, nodePoolNames []string) {
```

Comment:
Do you need to sort the names directly? Seems like you could build the array from the ground up like

```
orderedNodePoolNames := append([]string{}, c.nodePoolsTimedOut)
for name := range nodePoolCandidates {
  if !c.nodePoolsTimedOut.Has(name) {
    orderedNodePoolNames = append(orderedNodePoolNames, name)
  }
}
```